lms/djangoapps/courseware/tests/test_entrance_exam.py

```diff
@@ -206,6 +211,100 @@ def setUp(self):
             ]
         )
 
+    @mock.patch('xmodule.x_module.XModuleMixin.has_dynamic_children', mock.Mock(return_value='True'))
+    def test_view_redirect_if_entrance_exam_required(self):
+        """
+        Unit Test: if entrance exam is required. Should return a redirect.
+        """
+        url = reverse('courseware', kwargs={'course_id': unicode(self.course.id)})
+        expected_url = reverse('courseware_section',
+                               kwargs={
+                                   'course_id': unicode(self.course.id),
+                                   'chapter': self.entrance_exam.location.name,
+                                   'section': self.exam_1.location.name
+                               })
+        resp = self.client.get(url)
+        self.assertRedirects(resp, expected_url, status_code=302, target_status_code=200)
+
+    @patch.dict('django.conf.settings.FEATURES', {'ENTRANCE_EXAMS': False})
+    def test_entrance_exam_content_absence(self):
+        """
+        Unit Test: If entrance exam is not enabled then its content e.g. problems should not be loaded.
+        """
+        url = reverse('courseware', kwargs={'course_id': unicode(self.course.id)})
+        resp = self.client.get(url)
+        self.assertNotIn('Exam Problem - Problem 1', resp.content)
+        self.assertNotIn('Exam Problem - Problem 2', resp.content)
+
+    def test_entrance_exam_requirement_message(self):
+        """
+        Unit Test: entrance exam requirement message should be present in response
+        """
+        url = reverse(
+            'courseware_section',
+            kwargs={
+                'course_id': unicode(self.course.id),
+                'chapter': self.entrance_exam.location.name,
+                'section': self.exam_1.location.name
+            }
+        )
+        resp = self.client.get(url)
+        self.assertEqual(resp.status_code, 200)
```

Comment:
Test that "Exam Problem - Problem 1" and "Exam Problem - Problem 2" are present on the page. Currently there is no test point that ever verifies they are present when they should be present-- there is only a test point that verifies that they are absent in one case. That means that test_entrance_exam_content_absence could be passing incorrectly if the problems were never added with those exact names to the course.